### DISCUSSION TITLE
Local playback on the server hosting myMpd

### DISCUSSION THREAD

**Discussion by stratosgear** [jcorporation/myMPD#1233] [Q&A] [answered]
I am not sure if this is supported or not, and I have issues finding an answer in the docs.

My setup is a beefy server that hosts my music library and runs the mpd server.  It is configured with an audio_output of type httpd able to stream music. The server  is hosted outside my local network.

At home I have a raspberry pi 3, that has an IQAudio DigiDac+ and it has myMpd installed, no local mpd.  The dac outputs to an amplifier connected to a nice pair of speakers.  It is also accessible, from my local network on pi3.local.

Now I want to be able to connect to pi3.local, from within my local network, access myMpd, select music and played it "locally?" (is that what locally means?) on the pi3, and out through the speakers. If I access myMpd from another computer/location and change songs, the pi should switch to the new song.

I bet I have something misconfigured (or not configured at all) but I cannot find instructions on what to look where in order to fix my issue.... :(

I can see through aplay -L or -l that the dac is properly loaded/available on the pi, but there is no music output at all...

```
[root@pi3 state]# aplay -L
null
    Discard all samples (playback) or generate zero samples (capture)
default:CARD=IQaudIODAC
    IQaudIODAC, IQaudIO DAC HiFi pcm512x-hifi-0
    Default Audio Device
sysdefault:CARD=IQaudIODAC
    IQaudIODAC, IQaudIO DAC HiFi pcm512x-hifi-0
    Default Audio Device
[root@pi3 state]# aplay -l
**** List of PLAYBACK Hardware Devices ****
card 0: IQaudIODAC [IQaudIODAC], device 0: IQaudIO DAC HiFi pcm512x-hifi-0 [IQaudIO DAC HiFi pcm512x-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

[EDITED to add:]
I would expect my "perfect" setup would also support the same behavior when playing webradio stations.

Is there a kind soul that would describe the individual steps required to get this working, so I have something to go through to make sure everything is in place...?

A quick sidenote.  I had something similar working with Moode and/or Volumio (that was working) but I was not really happy with their interfaces and their overall project direction.  myMpd is the nicest experience that I would expect to use, but alas cannot make it work!

Thanks so much!

**Comment by jcorporation** [comment, marked as answer]
The local playback feature of myMPD utilizes the clients browser to play back a mpd http stream. myMPD can only proxy this stream.

The myMPD backend can not play back audio.

In your setup you can use the MPD snapcast output plugin and the snapcast client on your local pi or any other streaming solution.

As an alternative a MPD satellite setup could work: https://jcorporation.github.io/myMPD/additional-topics/mpd-satellite-setup

**Comment by stratosgear** [comment]
> The myMPD backend can not play back audio.

I do not believe I have read that sentence anywhere in the docs. :) That explains a lot.

And I also now understand why a satellite  setup might be beneficial!

Thanks!
